terraform/node_module_variable.go

```diff
@@ -137,57 +140,37 @@ func (n *nodeModuleVariable) ModulePath() addrs.Module {
 	return n.Addr.Module.Module()
 }
 
-// GraphNodeEvalable
-func (n *nodeModuleVariable) EvalTree() EvalNode {
+// GraphNodeExecutable
+func (n *nodeModuleVariable) Execute(ctx EvalContext, op walkOperation) error {
 	// If we have no value, do nothing
 	if n.Expr == nil {
-		return &EvalNoop{}
+		return nil
 	}
 
 	// Otherwise, interpolate the value of this variable and set it
 	// within the variables mapping.
-	vals := make(map[string]cty.Value)
+	var vals map[string]cty.Value
+	var err error
+
+	switch op {
+	case walkRefresh, walkPlan, walkApply, walkDestroy, walkImport:
```

Comment:
It's an entirely reasonable question, and I have no idea how to respond: It's _never_ clear to me when those filters matter or not - maybe this node never gets called during walkEval, so why check? I DUNNO!!! 

There are definitely places where I fear I'm copying logic that isn't actually helpful; I don't know why some nodes have filters and others don't. I imagine many of them are important, load-bearing `switch`es while some are leftovers from who knows how many years ago. 

This isn't a satisfying response, I know. Perhaps when the whole refactor is done I can make another pass through all of them and try and piece together which are important, what might be missing, etc.

Reply:
I assumed that this would require another pass too. As I understood it, the first step would be to get all the existing logic moved from eval nodes into `Execute` calls. We could then refactor that more easily, and in the process re-evaluate any of the other inconsistencies that may be in this code which has evolved quite a bit over time.

Reply:
that's the dream!